subiquity/models/subiquity.py

```diff
@@ -308,12 +326,25 @@ def _cloud_init_files(self):
             ('etc/cloud/cloud.cfg.d/99-installer.cfg', config, 0o600),
             ('etc/cloud/ds-identify.cfg', 'policy: enabled\n', 0o644),
             ]
+        # Add cloud-init clean hooks to support golden-image creation.
+        cfg_files = ["/" + path for (path, _content, _cmode) in files]
+        cfg_files.extend(self.network.rendered_config_paths())
```

Comment:
It's not necessarily something to fix in this branch, but this feels a bit tightly coupled. We shouldn't ideally have to change the code here if another model class emits config files that should be cleaned up. Other code does "for model in models: if hasattr(model, 'make_cloudconfig'): model.make_cloudconfig()" type games but that's also kind of ugly. A functools.singledispatch with implementations for the models might be nicer... not something for Chad to worry about though :-)

Reply:
@mwhudson I think Chad's role on this PR is more than complete, can we take these items as notes to address in a future PR?